### PR TITLE
validateProperties APIs

### PR DESCRIPTION
**Pull request by PatAltimore**

### CODE REVIEW

File: api-reference/beta/api/group_validateproperties.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# group: validateProperties
```
Comment (DeniseKwok) [inline api-reference/beta/api/group_validateproperties.md:1]: nit: Would it be useful to describe why one might use this API over the other one? It is used for Group update scenarios so it won't validate mailNickname uniqueness against its own mailNickname.

File: api-reference/beta/api/directoryobject_validateproperties.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+# directoryObject: validateProperties
+
+Validate an Office 365 group's display name or mail nickname complies with naming policies.  Clients can use the API to determine if a display name or mail nickname is valid before trying to create an Office 365 group. 
+
+The following validations are performed for the display name and mail nickname properties: 
+1. Validate the prefix and suffix naming policy
+2. Validate the custom banned words policy
+3. Validate the mail nickname is unique
```
Comment (DeniseKwok) [inline api-reference/beta/api/directoryobject_validateproperties.md]: Maybe we should clarify that it only checks uniqueness against groups with groupType "Unified"? I don't know if we already clarify that in the Group creation docs

File: api-reference/beta/api/directoryobject_validateproperties.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+## Request body
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter	   | Type	|Description|
+|:---------------|:--------|:----------|
+|entityType|String| `Group` is the only supported entity type. |
+|displayName|String| The display name of the group to validate. The property is not individually required. However, at least one property (displayName or mailNickname) is required. |
+|mailNickname|String| The mail nickname of the group to validate. The property is not individually required. However, at least one property (displayName or mailNickname) is required. |
+|onBehalfOfUserId|Guid| The object ID of the user to impersonate when calling the API. The validation results are for the onBehalfOfUserId's attributes and roles. |
```
Comment (dkershaw10) [inline api-reference/beta/api/directoryobject_validateproperties.md]: We should not say object ID.  This should just be `id`.

File: api-reference/beta/api/group_validateproperties.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+Validate if an Office 365 group's display name or mail nickname complies with naming policies. Clients can use the API to determine if a display name or mail nickname is valid before trying to **update** an Office 365 group. For validating properties before creating a group, use the [validateProperties function](directoryobject_validateproperties.md) for directory objects.
```
Comment (dkershaw10) [inline api-reference/beta/api/group_validateproperties.md]: Per Denise's comments - please remove references to mailNickName here.

File: api-reference/beta/api/group_validateproperties.md (hunk 1 above)
Comment (dkershaw10) [reply api-reference/beta/api/group_validateproperties.md:1]: I get that, but that seems like unnecessary detail to me.  I would prefer to just provide the guidance that prior to group updates to displayName or mailNickname, use this API to validate the displayName and mailNickName.

**Pull request closed without merging**